src/state/Data/actions.js

```diff
@@ -4,36 +4,467 @@ import attributeData from './AttributeData/actions';
+ * @param {bool} loadRelations Whether response should contain relations
+ * @param {Array?} dataSourceKeys Another optional filter parameter. Possible to use insted of layerTemplateKey or areaTreeKey
+ * @param {Array?} order 
+ * @param {Object} mergedSpatialFilter Filler object contains modifiers and layerTemplateKey or areaTreeLevelKey.
+ * @param {Object} mergedAttributeFilter Filler object contains modifiers, layerTemplateKey or areaTreeLevelKey and styleKey.
+ */
+function loadIndexedPage(styleKey, relations, featureKeys, spatialIndex, spatialFilter, attributeFilter, loadGeometry, loadRelations, dataSourceKeys, order, mergedSpatialFilter, mergedAttributeFilter) {
+	return (dispatch, getState) => {
+		const localConfig = Select.app.getCompleteLocalConfiguration(getState());
+		const apiPath = 'backend/rest/data/filtered';
+
+        const {areaTreeLevelKey, layerTemplateKey, ...modifiers} = mergedSpatialFilter
+        const usedRelations = relations ? {...relations} : DEFAULT_RELATIONS_PAGE;
+        
+        //register indexes
+
+        ////
+        // Spatial
+        ////
+        if(loadGeometry) {
+            const loadingTilesGeometry = spatialIndex?.tiles || spatialFilter.tiles;
+            dispatch(spatialData.addLoadingIndex(mergedSpatialFilter, order, spatialFilter.level, loadingTilesGeometry));
+        }
+
+        
+        ////
+        // Attribute
+        ////
+        const loadingTilesAttributes = spatialIndex?.tiles || spatialFilter.tiles;
+        dispatch(attributeData.addLoadingIndex(mergedAttributeFilter, order, spatialFilter.level, loadingTilesAttributes));
+
+
+        // Create payload
+
+		const payload = {
+            modifiers,
+
+            // which layer you want
+            ...(layerTemplateKey && {layerTemplateKey}),
+            ...(areaTreeLevelKey && {areaTreeLevelKey}),
+
+            // get attributes from style
+            ...(styleKey && {styleKey}),
+
+            // pagination for relations (& data sources)
+            relations: usedRelations,
+
+            data: {
+                // list of features you want
+                ...(featureKeys && {featureKeys}),
+
+                // which tiles you want (pseudo-pagination)
+                // spatialIndex: {
+                //     tiles: [[lon, lat], ...],
+                // },
+                ...(spatialIndex && {spatialIndex}),
+
+                // extent
+                spatialFilter,
+
+                // filter features by attribute value
+                // attributeFilter: {
+                //     'attribute-uuid': "blue",
+                //     'attribute-uuid': {
+                //         in: [12, 13]
+                //     },
+                //     ...
+                // },
+                ...(attributeFilter && {attributeFilter}),
+                
+                //request for geometry
+                geometry: loadGeometry === false ? false : true,
```

Comment:
You mean loadGeometry? It "should" be only boolean. 

I think think that loadGeometry === false ? false : true has same value like !!loadGeometry. You prefer !!loadGeometry?

Reply:
I think !!thing is simply readable as makeBoolean(thing), whereas the ternary makes me think about what's special here.
But if loadGeometry should be only boolean, then why is it not just "geometry: loadGeometry"?

Reply:
I can not guarantee that loadGeometry is boolean, it could be undefined, null or something else. Yes ```  geometry: loadGeometry``` is a possibility, but ```geometry: loadGeometry === false ? false : true,``` will sent to BE always bool.

Reply:
!!